bigml/ensemble.py

```diff
@@ -178,17 +181,40 @@ def __init__(self, ensemble, api=None, max_models=None, cache_get=None):
                         query_string=query_string,
                         no_check_fields=no_check_fields)
 
+                models = [model]
+
+        if self.distributions is None:
+            try:
+                self.distributions = [{'training': {'categories': m.tree.distribution}}
+                                      for m in models]
+            except AttributeError:
+                self.distributions = [m['object']['model']['distribution']
+                                      for m in models]
+
         if self.boosting is None:
             self._add_models_attrs(model, max_models)
+
         if self.fields is None:
             self.fields, self.objective_id = self.all_model_fields(
                 max_models=max_models)
-        if len(self.models_splits) == 1:
-            self.multi_model = MultiModel(models, self.api, fields=self.fields)
 
         self.regression = \
             self.fields[self.objective_id].get('optype') == 'numeric'
 
+        if not self.regression and self.boosting is None:
+            classes = set()
+            for distribution in self.distributions:
```

Comment:
Oh, I see... I'm paranoid on the download of big quantities of models... Is this code going to be run always?Independently of whether we use an ensemble to build the Ensemble or a list of models? maybe we could at least not do that when building from an ensemble...